src/mscorlib/src/System/Char.cs

```diff
@@ -905,20 +901,22 @@ public static double GetNumericValue(String s, int index)
         // Plane 16 contains 0x100000 ~ 0x10ffff.
         internal const int UNICODE_PLANE16_END   = 0x10ffff;
 
-        internal const int HIGH_SURROGATE_START = 0x00d800;
-        internal const int LOW_SURROGATE_END    = 0x00dfff;
+        internal const int HIGH_SURROGATE_START = 0xd800;
+        internal const int HIGH_SURROGATE_END = 0xdbff;
+        internal const int LOW_SURROGATE_START = 0xdc00; 
+        internal const int LOW_SURROGATE_END = 0xdfff;
         
 
 
        /*================================= ConvertFromUtf32 ============================
         ** Convert an UTF32 value into a surrogate pair.
         ==============================================================================*/
         
-        public static String ConvertFromUtf32(int utf32)
+        public unsafe static String ConvertFromUtf32(int utf32)
```

Comment:
I would, however, like the unsafe modifier scoped to the part of the method that needs it (specifically around the stackalloc).  We should strive to minimize the amount of unsafe regions and how much code is in an unsafe context.